docs/develop.md

```diff
@@ -1 +1,230 @@
+:::
+
+## Lifecycle Management (LCM)
+
+[`Renovate`](https://docs.renovatebot.com/) is used to keep your dependencies up to date.
+
+The templates by default make use of a centralized configuration we provide for RefArch-based projects. More information can be found in the [RefArch documentation](https://refarch.oss.muenchen.de/tools.html#renovate).
+
+To modify the default Renovate settings, the `renovate.json5` file in the project's root directory can be edited.
+
+::: info Information
+To make Renovate work make sure that it has access to your GitHub repository.
+For projects in the `it-at-m` organization Renovate automatically has access and is enabled when the configuration file is found in your repository.
+:::
+
+::: danger IMPORTANT
+To finish the onboarding process of Renovate, you need to open a PR for a dependency update found by Renovate through the "Dependency Dashboard" issue.
+This PR then has to be merged manually once.
+After that's done Renovate will start opening PRs normally.
+:::
+
+## Pull Requests
+
+When a pull request (PR) is created, several tools help maintain code quality:
+
+### Code Rabbit
+
+**Code Rabbit** is an AI-powered code reviewer that assists with PR assessments. The configuration file can be found at the root of the templates in `.coderabbit.yaml`.
+
+Our configuration enables automatic reviews (and follow-up reviews when changes to a PR have been made). Additionally, we set CodeRabbit in "nitpicky" mode to find all of those nasty bugs.
+Feel free to customize the configuration to your own needs. More information is available in the [official documentation](https://docs.coderabbit.ai/).
+
+::: info Information
+To make CodeRabbit work make sure that it has access to your GitHub repository.
+For projects in the `it-at-m` organization CodeRabbit automatically has access and is enabled when the configuration file is found in your repository.
+:::
+
+### CodeQL
+
+**CodeQL** is a GitHub tool for discovering vulnerabilities and code smells in code. More details can be found [here](https://codeql.github.com/).
+
+The template enables CodeQL for Pull Requests and configures CodeQL to only scan for Java and JavaScript/TypeScript/Vue files by default.
+For further information on how to change the configuration, please check out the documentation of the related custom [GitHub workflow](https://github.com/it-at-m/.github/blob/main/workflow-templates/codeql.yaml).
+
+::: danger IMPORTANT
+If you are using Java-based projects inside your repository, you need to add those to the `java-build-path` variable pointing to the directory of the `pom.xml` files.
+:::
+
+### Dependency Review
+
+To ensure that only dependencies with approved licenses are included, a [global check](https://github.com/it-at-m/.github/blob/main/workflow-configs/dependency_review.yaml) is implemented.
```

Comment:
Reference to [GitHub Dependency Review](https://docs.github.com/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review) might be good